source/adapters/cuda/enqueue.cpp

```diff
@@ -284,9 +284,15 @@ setKernelParams(const ur_context_handle_t Context,
                                    CudaImplicitOffset);
     }
 
-    if (Context->getDevice()->maxLocalMemSizeChosen()) {
+    auto Device = Context->getDevice();
+    if (LocalSize > static_cast<uint32_t>(Device->getMaxCapacityLocalMem())) {
+      setErrorMessage("Too much local memory allocated for device",
```

Comment:
I'd change the message with something like, `Excessive allocation of local memory on the device`

Reply:
Ok, that does sound more suitable